packages/schema/lib/schemas/AuthFieldSchema.js

```diff
@@ -0,0 +1,90 @@
+'use strict';
+
+const makeSchema = require('../utils/makeSchema');
+const FieldSchema = require('./FieldSchema');
+
+module.exports = makeSchema(
+  {
+    ...FieldSchema.schema,
+    id: '/AuthFieldSchema',
+    description: 'Field schema specialized for authentication fields.',
+    properties: {
+      ...FieldSchema.schema.properties,
+      isSafe: {
+        description:
+          'Indicates if this authentication field is safe (not secret).',
+        type: 'boolean',
+      },
+    },
+
+    // Do not allow certain isSafe = True if sensitive tokens exists
+    not: {
+      properties: {
+        key: {
+          enum: [
+            'access_token',
+            'refresh_token',
+            'api_key',
+            'password',
+            'secret',
+          ],
+        },
+        isSafe: { const: true },
+      },
+      required: ['key', 'isSafe'],
+    },
+    // Add examples and anti-examples specifically for basic & custom auth
+    examples: [
+      // Basic Auth - email & password
+      {
+        key: 'email',
+        type: 'string',
+        isSafe: true,
+        required: true,
+      },
+      {
+        key: 'password',
+        type: 'string',
+        isSafe: false,
+        required: true,
+      },
+
+      // Custom Auth - api key
+      {
+        key: 'api_key',
+        type: 'string',
+        isSafe: false,
+        required: true,
+      },
+    ],
+
+    antiExamples: [
+      {
+        example: {
+          key: 'password',
+          type: 'string',
+          isSafe: true,
+          required: true,
+        },
+        reason: 'A "password" field cannot have isSafe = true.',
+      },
+      {
+        example: {
+          key: 'api_key',
+          isSafe: true,
+        },
+        reason:
+          'Missing "type". Also "api_key" is a sensitive field and cannot have isSafe=true.',
```

Comment:
```suggestion
          '"api_key" is a sensitive field and cannot have isSafe set as true.',
```

The `key` is the only required property of a field https://github.com/zapier/zapier-platform/blob/main/packages/schema/lib/schemas/FieldSchema.js#L27